nautobot/extras/tests/test_customfields.py

```diff
@@ -157,6 +157,52 @@ def test_multi_select_field(self):
         # Delete the custom field
         cf.delete()
 
+    def test_text_field_value(self):
+        obj_type = ContentType.objects.get_for_model(Site)
+
+        # Create a custom field
+        cf = CustomField(
+            type=CustomFieldTypeChoices.TYPE_TEXT,
+            name="my_text_field",
+            required=False,
+        )
+        cf.save()
+        cf.content_types.set([obj_type])
+
+        # Assign a disallowed value (list) to the first Site
+        site = Site.objects.first()
+        site.cf[cf.name] = ["I", "am", "a", "list"]
+        try:
+            site.validated_save()
+            self.fail("Custom fields with a type of Text should not be able to save a list value")
+        except ValidationError as e:
+            self.assertIn("Value must be a string", str(e))
```

Comment:
You could also use `assertRaises` for each of these checks, e.g.:

```suggestion
        with self.assertRaises(ValidationError) as context:
            site.validated_save()
        self.assertIn("Value must be a string", str(context.exception))
```